### PR TITLE
#355 Retrieve the subText property from links

### PR DESCRIPTION
**Pull request by katarinazaprazna**

### CODE REVIEW

File: next/src/utils/useGetLinkProps.ts [hunk 1]
```diff
@@ -58,8 +56,27 @@ export const useGetLinkProps = () => {
...
+    } else if (link?.workshop?.data?.attributes && 'dates' in link.workshop.data.attributes) {
+      subText = formatMostRecentWorkshopDate(link.workshop.data)
+    }
+
+    return {
+      label,
+      href,
+      target,
+      subText,
+    }
```
Comment (radoslavzeman) [inline next/src/utils/useGetLinkProps.ts:78]: I don't think this should be part of this hook.

Getting links props (href, label, target) and getting subtext are to separate non-related "tasks", even though both are displayed on the same Card component. 

Similarly, we do not fetch article tag (or perex) through `useGetLinkProps` when displayed for example in carousel:
image

Here I see that `getAdditionalLinkProps` mixes these two tasks.

**Review by radoslavzeman — CHANGES_REQUESTED** [review]

File: next/src/utils/useGetLinkProps.ts (hunk 1 above)
Comment (katarinazaprazna) [reply next/src/utils/useGetLinkProps.ts:78]: I'll check how articles are being handled!

File: next/src/components/common/NavBar/NavMenu/NavMenuSectionLink.tsx [hunk 2]
```diff
@@ -16,20 +18,35 @@ type NavMenuSectionLinkProps = {
 
 const NavMenuSectionLink = ({ link, divider }: NavMenuSectionLinkProps) => {
   const { getLinkProps } = useGetLinkProps()
-  const { children, href, subText, target } = getLinkProps(link)
+  const { t } = useTranslation()
+  const { children, href, target } = getLinkProps(link)
+  const mostRecentWorkshopDate =
+    formatMostRecentWorkshopDate(link?.workshop?.data?.attributes?.dates) ?? ''
```
Comment (radoslavzeman) [inline next/src/components/common/NavBar/NavMenu/NavMenuSectionLink.tsx:24]: Just for clean code, I would write `?? null`

File: next/src/components/common/NavBar/NavMenu/NavMenuSectionLink.tsx [hunk 3]
```diff
@@ -16,20 +18,35 @@ type NavMenuSectionLinkProps = {
...
 
   return isDefined(link.workshop?.data) ? (
     <>
       {divider}
       <li key={link.id}>
-        <MenuItemWorkshopCard title={children ?? ''} subText="" linkHref={href} />
+        <MenuItemWorkshopCard
+          title={children ?? ''}
```
Comment (radoslavzeman) [inline next/src/components/common/NavBar/NavMenu/NavMenuSectionLink.tsx:31]: `children` are always string, so there is no need of `?? ''`

File: next/src/components/common/NavBar/NavMenu/NavMenuSectionLink.tsx [hunk 4]
```diff
@@ -16,20 +18,35 @@ type NavMenuSectionLinkProps = {
...
+        />
       </li>
     </>
   ) : isDefined(link.branch?.data) ? (
     <>
       {divider}
       <li key={link.id}>
-        <MenuItemBranchCard title={children ?? ''} subText={subText ?? ''} linkHref={href} />
+        <MenuItemBranchCard
+          title={children ?? ''}
```
Comment (radoslavzeman) [inline next/src/components/common/NavBar/NavMenu/NavMenuSectionLink.tsx:46]: `children` are always string, so there is no need of `?? ''`

File: next/src/utils/formatMostRecentWorkshopDate.ts [hunk 5]
```diff
@@ -1,8 +1,13 @@
-import { WorkshopSlugEntityFragment } from '@/src/services/graphql/api'
+type WorkshopDates =
```
[the remark this answers is not in the document]
Comment (radoslavzeman) [reply next/src/utils/formatMostRecentWorkshopDate.ts:1]: You can create for example `WorkhopDateItem` fragment in graphql file and use the generated frangment type (`WorkshopDateItemFragment`).

Anyway, I think that Strapi types datetimes as `any` unfortunately :/ (Or tell me if I am wrong) But still, this suggested approach would be nice to follow.

```graphql
fragment WorkshopDateItem on ComponentItemsWorkshopDate {
  datetime
}

fragment WorkshopSlugEntity on WorkshopEntity {
  __typename
  id
  attributes {
    title
    slug
    dates {
      ...WorkshopDateItem
    }
  }
}
```

**Review by radoslavzeman — APPROVED** [review]

**Pull request merged**